### PR TITLE
SW-2864 Region category under My Account

### PR DESCRIPTION
**Pull request by constanzauanini**
Screenshot 2023-02-03 at 12 58 03

Didn't add language, since language will remain on left menu for now

### CODE REVIEW

**Comment by karthikbtf** [comment]
> Screenshot 2023-02-03 at 12 58 03
> 
> Didn't add language, since language will remain on left menu for now

I think Ming wants it under My Account as well, may need to check with her?

**Review by karthikbtf — APPROVED** [review]
Looks good!! In mobile too!
Maybe check with Ming about language.. wondering if we should just remove the tooltip for time zone too. We can follow up separately for sure.

**Comment by constanzauanini** [comment]
@karthikbtf, Ming said we wanted locale selection in both places. I'm having trouble updating the locale from my account. It gets updated for a second but then it changes back. Any idea of why could this be hapening? Maybe related to LocalizationContext.Consumer? CC @sgrimm

**Comment by karthikbtf** [comment]
> @karthikbtf, Ming said we wanted locale selection in both places. I'm having trouble updating the locale from my account. It gets updated for a second but then it changes back. Any idea of why could this be hapening? Maybe related to LocalizationContext.Consumer? CC @sgrimm

@constanzauanini yes.. i pointed out why in the review.. it's a simple fix

File: src/components/MyAccount/index.tsx [hunk 1]
```diff
@@ -156,6 +161,8 @@ const MyAccountContent = ({
       setPersonOrganizations(addRoleNames(organizations));
     }
     setRemovedOrg(undefined);
+    setPreferredWeightSystemSelected((userPreferences?.preferredWeightSystem as string) || 'metric');
```
[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/components/MyAccount/index.tsx:164]: ignore ^^ it was a PR display issue

File: src/components/MyAccount/index.tsx [hunk 2]
```diff
@@ -183,6 +190,10 @@ const MyAccountContent = ({
         await PreferencesService.updateUserPreferences({ preferredWeightSystem: preferredWeightSystemSelected });
         reloadUserPreferences();
       }
+      if (localeSelectionEnabled) {
+        await UserService.updateUser({ ...user, locale: localeSelected });
```
[the remark this answers is not in the document]
Comment (constanzauanini) [reply src/components/MyAccount/index.tsx:194]: Thank you! That was the problem!

File: src/strings/strings-en.ts [hunk 3]
```diff
@@ -403,6 +403,8 @@ export const strings = {
   LAB_GERMINATION: 'Lab Germination',
   LAB: 'Lab',
   LANDOWNER: 'Landowner',
+  LANGUAGE: 'Language',
+  LANGUAGE_AND_REGION: 'Language and Region',
```
[the remark this answers is not in the document]
Comment (constanzauanini) [reply src/strings/strings-en.ts:407]: Hmm, I think it's ok, it will there in the future. I can change it to be just region if we prefer

**Review by karthikbtf — APPROVED** [review]
looks good in vercel

**Pull request merged**